presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/execution/ExceptionUtils.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.functionNamespace.execution;
+
+import com.facebook.drift.TException;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.StandardErrorCode;
+import com.facebook.presto.thrift.api.udf.ThriftUdfServiceException;
+import com.facebook.presto.thrift.api.udf.UdfExecutionFailureInfo;
+
+import javax.annotation.Nullable;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+
+public class ExceptionUtils
+{
+    private static final Pattern STACK_TRACE_PATTERN = Pattern.compile("(.*)\\.(.*)\\(([^:]*)(?::(.*))?\\)");
+
+    private ExceptionUtils()
+    {
+    }
+
+    public static PrestoException toPrestoException(ThriftUdfServiceException thriftException)
+    {
+        UdfExecutionException cause = toUdfExecutionException(thriftException.getFailureInfo());
+        PrestoException prestoException = new PrestoException(
+                thriftException::getErrorCode,
+                firstNonNull(cause.getMessage(), "Error executing remote function"),
```

Comment:
Seems like `toUdfExecutionException` can return `null` so you can potentially get a NPE here? Maybe should use `Optional`.